testing/functional_tests/test_automate_training.py

```diff
@@ -34,6 +35,38 @@ def test_automate_training(download_functional_test_files, script_runner):
     check_sha256(str(file_config))
 
 
+def test_automate_training_run_test_debug(download_functional_test_files):
+    """A unit test similar to test_automate_training_run_test but allow step through (instead of using script caller/
```

Comment:
Happy to elaborate on this. So what happened is for "subprocess" version of the unit test above this current one, it spawn a system subprocess to carry out the CLI based action (invoke the automated training in this case, the test was designed to test a CLI command after all, so make sense to do that). The problem with that is, while it can conduct [black box testing](https://en.wikipedia.org/wiki/Black-box_testing), it does NOT allow debug/stepping through the code (also known as [white-box testing](https://en.wikipedia.org/wiki/White-box_testing)). `Automated Training` test being a very high level test, it frequently fails during development just because it covers such a huge area of the code. If it fails, it is actually very hard to figure out what goes wrong when you cannot step through the code. 

Hence, why adding this test to really help development process actually.